targets/mbed/source/jerry_run.cpp

```diff
@@ -17,109 +17,100 @@
 
 int js_entry (const char *source_p, const size_t source_size)
 {
-  const jerry_api_char_t *jerry_src = (const jerry_api_char_t *) source_p;
-  jerry_completion_code_t ret_code = JERRY_COMPLETION_CODE_OK;
-  jerry_flag_t flags = JERRY_FLAG_EMPTY;
-  jerry_api_object_t *err_obj_p = NULL;
-  jerry_api_value_t err_value = jerry_api_create_void_value ();
-
-  jerry_init (flags);
+  const jerry_char_t *jerry_src = (const jerry_char_t *) source_p;
+  jerry_init (JERRY_INIT_EMPTY);
+  uint8_t ret_code = 0;
 
   js_register_functions ();
 
-  if (!jerry_parse (jerry_src, source_size, &err_obj_p))
+  jerry_value_t parsed_code = jerry_parse (jerry_src, source_size, false);
+
+  if (!jerry_value_has_error_flag (parsed_code))
   {
-    printf ("Error: jerry_parse failed\r\n");
-    ret_code = JERRY_COMPLETION_CODE_UNHANDLED_EXCEPTION;
-    jerry_api_release_object (err_obj_p);
+    jerry_value_t ret_value = jerry_create_undefined ();
+    ret_value = jerry_run (parsed_code);
+    jerry_release_value (ret_value);
   }
   else
   {
-    if ((flags & JERRY_FLAG_PARSE_ONLY) == 0)
-    {
-      ret_code = jerry_run (&err_value);
-      jerry_api_string_t *err_str_p = NULL;
-
-      if (err_str_p != NULL)
-      {
-        jerry_api_release_string (err_str_p);
-      }
-    }
+    printf("Error: jerry_parse failed!\r\n");
+    ret_code = -1;
   }
 
+  jerry_release_value (parsed_code);
+
   return ret_code;
 }
 
 int js_eval (const char *source_p, const size_t source_size)
 {
-  jerry_completion_code_t status;
-  jerry_api_value_t res;
+  int status = 0;
 
-  status = jerry_api_eval ((jerry_api_char_t *) source_p,
-                           source_size,
-                           false,
-                           false,
-                           &res);
+  jerry_value_t ret_val = jerry_eval ((jerry_char_t *) source_p,
+                        source_size,
+                        false);
 
-  jerry_api_release_value (&res);
+  if (jerry_value_has_error_flag (ret_val))
+  {
+    printf("Error: jerry_eval failed!\r\n");
+    status = -1;
+  }
+
+  jerry_release_value (ret_val);
 
   return status;
 }
 
 int js_loop (uint32_t ticknow)
 {
-  jerry_api_object_t *global_obj_p;
-  jerry_api_value_t sysloop_func;
-  bool is_ok;
-
-  global_obj_p = jerry_api_get_global ();
-  is_ok = jerry_api_get_object_field_value (global_obj_p,
-                          (const jerry_api_char_t*) fn_sys_loop_name,
-                          &sysloop_func);
-  if (!is_ok)
+  int status = 0;
+  jerry_value_t global_obj = jerry_get_global_object ();
+  jerry_value_t sys_name = jerry_create_string ((const jerry_char_t *) fn_sys_loop_name);
+  jerry_value_t sysloop_func = jerry_get_property (global_obj, sys_name);
+  
+  jerry_release_value (sys_name);
+
+  if (jerry_value_has_error_flag (sysloop_func))
   {
     printf ("Error: '%s' not defined!!!\r\n", fn_sys_loop_name);
-    jerry_api_release_object (global_obj_p);
-    return -1;
+    status = -1;
```

Comment:
There is no return after an error? Are you sure this is correct? But don't forget to free values if you put a return here.